### PR TITLE
Create program stub for LS command

### PR DESCRIPTION
**Pull request by dreamer**
```
    Add LS program stub to replace LS command
    
    Making LS a program prevents the problem with invoking programs called
    ls.com or ls.exe - one game, that has such executable name is
    Links LS 1997.
    
    This change introduces a small "regression" - command parameters cannot
    be passed without space, as for DOS intenal commands, e.g.:
    
      dir.txt - equivalent of dir *.txt (usually used for e.g. dir/w)
      ls.txt - does not work any more - but works exactly like GNU ls, the
               first parameter is passed after a space.
    
    The stub invokes existing LS command implementation (as LS shares some
    code with DIR command).  When shared code will be made
    public/non-static, the rest of LS command implementation will be moved
    to a new program file.
```
Fixes: #504

### CODE REVIEW

**Comment by Wengier** [comment]
@dreamer Thanks for creating the program stub for LS command. I think I did the similar approach for DOSBox-X earlier, but I am glad to see it is now being worked for DOSBox Staging too. I appreciate the effort!

File: src/dos/dos_programs.cpp [hunk 1]
```diff
@@ -1895,19 +1896,18 @@ void DOS_SetupPrograms(void) {
...
+	PROGRAMS_MakeFile("INTRO.COM", INTRO_ProgramStart);
+	PROGRAMS_MakeFile("LOADFIX.COM", LOADFIX_ProgramStart);
+	PROGRAMS_MakeFile("LS.COM", LS_ProgramStart);
+	PROGRAMS_MakeFile("MEM.COM", MEM_ProgramStart);
+	PROGRAMS_MakeFile("MOUNT.COM", MOUNT_ProgramStart);
+	PROGRAMS_MakeFile("RESCAN.COM", RESCAN_ProgramStart);
 	PROGRAMS_MakeFile("IMGMOUNT.COM", IMGMOUNT_ProgramStart);
 	PROGRAMS_MakeFile("KEYB.COM", KEYB_ProgramStart);
-
+	PROGRAMS_MakeFile("LOADROM.COM", LOADROM_ProgramStart);
```
Comment (kcgen) [inline src/dos/dos_programs.cpp:1912]: I followed the cleaned up sort order for BOOT.COM through RESCAN.COM, but LOADROM.COM didn't make it into the L's. (Maybe a technical reason?)

File: src/dos/program_ls.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#include "program_ls.h"
+
+#include <string>
+
+#include "shell.h"
+#include "support.h"
+
+void LS::Run()
+{
```
Comment (kcgen) [inline src/dos/program_ls.cpp:29]: Is `ChangeToLongCmd();` needed to allow long command lines? 
(More a question, I'm not sure what criteria necessitates it; or why it's simply not always enabled)

**Review by kcgen — APPROVED** [review]
Very tidy @dreamer. I've only got a couple questions.

File: src/dos/dos_programs.cpp (hunk 1 above)
Comment (dreamer) [reply src/dos/dos_programs.cpp]: Ah, no - just a mistake on my part :) I probably selected wrong lines for sorting in vim.

File: src/dos/program_ls.cpp (hunk 2 above)
Comment (dreamer) [reply src/dos/program_ls.cpp:29]: I'm not sure why `ChangeToLongCmd` method has some stuff related to `-secure` bundled in…

I think this function is effectively needed when dealing with paths on host machine, which can be long. In `ls` we deal only with paths and search patterns for virtual drives, so I don't think it's necessary.

Usage in `AUTOTYPE` is probably ok, as we might want to handle quite long key sequences in there.

**Review by kcgen — APPROVED** [review]
Thanks for the quick answers @dreamer.
All set from my side.

**Pull request merged**